core/src/Cuda/Kokkos_Cuda_Instance.hpp

```diff
@@ -160,8 +161,9 @@ class CudaInternal {
         m_scratchUnified(0),
         m_scratchFunctor(0),
         m_scratchConcurrentBitset(0),
-        m_stream(0) {}
-
+        m_stream(0),
+        m_instanceId(0) {}
+  uint32_t get_instance_id() const;
```

Comment:
Is it always going to return zero?  Do you anticipate to change this?

Reply:
Right now I'm just adding the "device type" part of the ID. Once we start working more with distinguishable instance copies I'll look at differentiating those instances

Reply:
(To answer the questions, "yes" and "eventually" respectively)

Reply:
Then why do you bother already adding the data member?  Why don't you just return 0 from `get_instance_id()`?

Reply:
Why don't we discuss at the meeting? Christian and I had a discussion in which we decided the instance ID should exist. I went to implement it, and realized we don't really do much with differentiated instances of Execution Spaces (perhaps modulo Cuda+Streams), and then waffled on whether I wanted the instance variable or not.

I like the idea of it just returning zero now, and then when we start differentiating logical instances, maybe add back in a more meaningful "logical id"

Reply:
@dalg24 @crtrott obviously we didn't decide this at the meeting. How do we feel about leaving out the logical ID for the moment, in most other places I don't see us differentiating instances

Reply:
My specific worry is that in different spaces it could mean different things... in Cuda it might make sense for two instances created from the same stream to have the same logical ID, where in others the only differentiator might be from different objects.

I'd like to have a better sense of what a logical ID is before baking it into the code this tightly